### PR TITLE
Changed readme links to relative

### PR DESCRIPTION
**Pull request by MarcinKonowalczyk**
While looking at issue #541, I've noticed that locally built documentation has hyperlinks to the online one. While this works when documentation is online, it's not ideal for offline build. Moreover, when browsing legacy documentation online, the links take one to the `stable` documentation and, arguably, they shouldn't. This is a minor issue, so I thought I'll just submit a PR and see what happens.

Sphinx's linkcheck is struggling a bit with the relative links (probably because of the order in which the pages are generated. see [this](https://github.com/sphinx-doc/sphinx/issues/9383) issue), so I've selectivelly silenced them.

I've also taken the liberty to make the .rst table a bit less wide.

### CODE REVIEW

**Comment by MarcinKonowalczyk** [comment]
I just realised that this means links on GitHub might not work quite correctly. I'll leave the PR open for review for a time being though.

**Comment by bbayles** [comment]
Yeah, I'd like the GitHub page to keep working. If there's a way to make them both work, I'm all for it. Please file an issue?

**Review by bbayles — CHANGES_REQUESTED** [review]
Sadly this will make the GitHub front page not work.

**Review by MarcinKonowalczyk — COMMENTED** [review]
How about a preprocessor step in conf.py (see new commits) ?

**Comment by MarcinKonowalczyk** [comment]
git into a bit of a fight with GitHub workflow. sorry, not something I could have fixed locally. turns out that 1) when creating files with python they *need* to be opened `open(..., 'w+')` mode and 2) the entire directory tree must already exist.

File: docs/conf.py [hunk 1]
```diff
@@ -21,6 +21,39 @@
...
+
+import re
+
+with open('../README.rst', 'r+') as source:
+    readme_file = source.readlines()
+
+# Make sure build directory exists. This is required for github workflow.
+build_dir = '_build'
+if not os.path.exists(build_dir):
+    os.makedirs(build_dir)
```
Comment (bbayles) [inline docs/conf.py:34]: Can this be `os.makedirs(build_dir, exist_ok=True)`?

File: docs/conf.py [hunk 2]
```diff
@@ -21,6 +21,39 @@
...
+
+# Change absolute links in README.rst to relative ones. This way the online
+# documentation copmiles with relative links, while README.rst on GitHub can
+# have absolute links. See issue #551.
+root_path = 'https://more-itertools.readthedocs.io/en/stable/'
+in_table = False
+with open(os.path.join('.', build_dir, 'README.pprst'), 'w+') as target:
```
Comment (bbayles) [inline docs/conf.py:41]: Why is the `w+` needed here?

File: docs/conf.py (hunk 1 above)
Comment (MarcinKonowalczyk) [reply docs/conf.py]: yup

File: docs/conf.py (hunk 2 above)
Comment (MarcinKonowalczyk) [reply docs/conf.py]: apparently it's not. I've changed it to 'w'

**Pull request merged**